pf_driver/src/ros/ros_main.cpp

```diff
@@ -59,7 +59,13 @@ int main(int argc, char* argv[])
     ROS_ERROR("Unable to start scan");
     return -1;
   }
-  ros::spin();
+
+  ros::Rate rate(10);
+  while (ros::ok() && pf_interface.is_scanner_accessible_) {
+      ros::spinOnce();
+      rate.sleep();
```

Comment:
not sure what's going on here - manually slowing down the ROS loop by sleeps seems very dangerous. The rate at which you handle events might be lower than the rate of incoming events (e.g. if you subscribe to a topic).

if you want to stop the node once `pf_interface.is_scanner_accesslibe_` becomes `false`, you should instead call `ros::shutdown()` in the place where `is_scanner_accessible_` is set to `false` (it's thread-safe, so no worries).